xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -894,6 +894,10 @@ bool CPVRChannelGroup::Persist(void)
   if (!HasChanges() || (!m_bLoaded && m_iGroupId != -1))
     return bReturn;
 
+  // Mark newly created groups as loaded so future updates will also be persisted...
+  if (m_iGroupId == -1)
+    m_bLoaded = true;
```

Comment:
How about initialize m_bLoaded with true in the ctors if they set m_iGroupId to -1?

https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/channels/PVRChannelGroup.cpp#L53
https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/channels/PVRChannelGroup.cpp#L88

Maybe also here, if iGroupId param can be -1:
https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/channels/PVRChannelGroup.cpp#L68

Or does this have any bad side effects?